service/kas/access/rewrap.go

```diff
@@ -141,16 +167,55 @@ func verifySignedRequestToken(ctx context.Context, in *kaspb.RewrapRequest) (*Re
 	switch publicKey := clientPublicKey.(type) {
 	case *rsa.PublicKey:
 		requestBody.PublicKey = publicKey
-		return requestBody, nil
+		return &requestBody, nil
 	case *ecdsa.PublicKey:
 		requestBody.PublicKey = publicKey
-		return requestBody, nil
+		return &requestBody, nil
 	default:
 		slog.WarnContext(ctx, fmt.Sprintf("clientPublicKey not a supported key, was [%T]", clientPublicKey))
 		return nil, err400("clientPublicKey unsupported type")
 	}
 }
 
+func verifySRT(ctx context.Context, srt string, dpopJWK jwk.Key) (string, error) {
+	v, err := jws.NewVerifier(jwa.RS256)
+	if err != nil {
+		slog.ErrorContext(ctx, "unable to load RSA verifier", "err", err)
+		return "", err500("unable to verify token")
+	}
+	a := strings.IndexByte(srt, '.')
+	b := strings.LastIndexByte(srt, '.')
+	if a < 0 || b <= a {
+		slog.ErrorContext(ctx, "failed to decode srt", "err", err)
+		return "", err400("encoding error")
+	}
+	unsigned := srt[:b]
+	decoder := base64.URLEncoding.WithPadding(base64.NoPadding)
+	sig, err := decoder.DecodeString(srt[b+1:])
+	if err != nil {
+		slog.ErrorContext(ctx, "failed to decode signature", "err", err)
+		return "", err400("signed content failure")
+	}
+
+	err = v.Verify([]byte(unsigned), sig, dpopJWK)
+	if err != nil {
+		slog.WarnContext(ctx, "unable to verify request token", "err", err, "srt", srt, "jwk", dpopJWK)
+		return "", err401("unable to verify request token")
+	}
+	payload, err := decoder.DecodeString(srt[a+1 : b])
+	if err != nil {
+		slog.ErrorContext(ctx, "failed to decode payload", "srt", srt, "err", err)
+		return "", err400("signed content failure")
+	}
+	var srtDecoded SignedRequestBody
+	err = json.Unmarshal(payload, &srtDecoded)
+	if err != nil {
+		slog.WarnContext(ctx, "unable to unmarshall srt", "err", err, "srt", srt, "jwk", dpopJWK)
+		return "", err400("signed content failure")
+	}
+	return srtDecoded.RequestBody, nil
```

Comment:
The manual parsing and verification is surprising to me. Could we have a comment to understand why we can't verify the JWT using a higher level API?

Reply:
reverted to using the higher level API now that i have the correct options